app/controllers/manage_proposal_controller.rb

```diff
@@ -70,4 +91,42 @@ def user_has_approver_permissions?
   def unauthorized?(response)
     response.status == 401
   end
+
+  def publish_delete_proposal(proposal, provider)
+    search_response = cmr_client.get_collections({ 'native_id': proposal['native_id'] }, token)
+
+    if search_response.body['hits'].to_s != '1'
+      # If the search has more than one hit or 0 hits, the record was not
+      # uniquely identified from it's native ID.
+      flash[:error] = I18n.t('controllers.manage_proposals.publish.flash.delete.not_found_error')
+      Rails.logger.info("Could not complete delete request from proposal with short name: #{proposal['short_name']} and id: #{proposal['id']} because it could not be located.")
+    elsif !search_response.body['items'][0]['meta']['granule-count'].zero?
+      # Do not allow the deletion of collections which have granules
+      flash[:error] = I18n.t('controllers.manage_proposals.publish.flash.delete.granules_error')
+      Rails.logger.info("Could not complete delete request from proposal with short name: #{proposal['short_name']} and id: #{proposal['id']} because it had granules at the time of the delete attempt.")
+    else
+      cmr_response = cmr_client.delete_collection(provider, proposal['native_id'], token)
+
+      if cmr_response.success?
+        flash[:success] = I18n.t('controllers.manage_proposals.publish.flash.delete.success')
+        Rails.logger.info("Proposal with short name: #{proposal['short_name']} and id: #{proposal['id']} were successfully deleted in the CMR")
+      else
+        flash[:error] = I18n.t('controllers.manage_proposals.publish.flash.delete.error')
+        Rails.logger.info("Proposal with short name: #{proposal['short_name']} and id: #{proposal['id']} could not be deleted in the CMR")
+      end
+    end
+  end
+
+  def publish_create_or_update_proposal(proposal, provider)
```

Comment:
Same comment about logging and flash messages. With this I would refer to https://github.com/nasa/mmt/blob/1d0064d9492f56be0bd03485b49b6deedc7814a1/app/controllers/collection_drafts_controller.rb#L95 (and the update action above) and https://github.com/nasa/mmt/blob/1d0064d9492f56be0bd03485b49b6deedc7814a1/app/controllers/base_published_record_controller.rb#L37 (can't seem to find the update action at the moment)